WS-004-modified register page

CadfanWilliams opened the pull request.

CadfanWilliams committed:
modified register page

AndrewYHuang commented on src/main/resources/templates/register.html:
```diff
@@ -2,32 +2,34 @@
+        </div>
+    </form>
+</section>
+</body>
```
Could we make sure that formatting and indentation is consistent - it'd make diffs a whole lot less noisy

AndrewYHuang commented on src/main/resources/templates/register.html:
```diff
@@ -2,32 +2,34 @@
+        <div class="input-container">
+            <div class="register-inputs">
+                <label for="username"> Username </label>
+                <input class="register-textfields" height="100%" id="username" type="text" name="username" th:field="*{username}"/>
```
What is this height attribute doing here?

AndrewYHuang commented on src/main/resources/templates/register.html:
```diff
@@ -2,32 +2,34 @@
-    </body>
+            <div class="register-inputs">
+                <label> Password  </label>
+                <input class="register-textfields" height="100%" type="password" name="password" th:field="*{password}"/>
```
What is this height attribute doing here?

AndrewYHuang commented on src/main/resources/templates/register.html:
```diff
@@ -2,32 +2,34 @@
+                <input class="register-textfields" height="100%" type="password" name="password" th:field="*{password}"/>
+            </div>
+            <div class="register-inputs">
+                <input class="register-textfields" width="80%" align="right" type="submit" value="Submit"/>
```
What are these width and align attributes doing here?

AndrewYHuang commented on src/main/resources/static/css/application.css:
```diff
@@ -45,8 +45,50 @@
+}
+.register-textfields{
+height:2rem;
+}
```
Could we make sure that formatting and indentation is consistent?

AndrewYHuang commented on src/main/resources/static/css/application.css:
```diff
@@ -45,8 +45,50 @@
+height:2rem;
+}
+
+
+
+
+
```
Unused newlines

AndrewYHuang commented on src/main/resources/static/css/application.css:
```diff
@@ -45,8 +45,50 @@
     display: flex;
     flex-direction: column;
-
+    height:100%;
```
This `height: 100%` isn't actually doing that much!

AndrewYHuang replied:
This one probably still needs removal - unless you have a reason not to! :)

AndrewYHuang commented on src/main/resources/static/css/application.css:
```diff
@@ -45,8 +45,50 @@
+    display: flex;
+    justify-content:center;
+   font-size: 30px;
+   height: 200%;
```
This `height: 200%` isn't really doing anything either!

Eyrim commented on src/main/resources/static/css/application.css:
```diff
@@ -45,8 +45,50 @@
+.register-form {
+    display: flex;
+    justify-content:center;
+   font-size: 30px;
```
Inconsistent indenting

AndrewYHuang commented on src/main/resources/static/css/application.css:
```diff
@@ -45,8 +45,50 @@
+}
+
+.register-body{
+height:1000px;
```
`height` doesn't actually do much for us here either, because elements generally automatically expand vertically to contain its contents

AndrewYHuang commented on src/main/resources/templates/register.html:
```diff
@@ -2,32 +2,34 @@
+<head>
+    <title>Register</title>
+</head>
+<body class="register-body">
```
Why have we removed the `page-container` class? Have you noticed that it sets `margin: 0;`? 
What do you notice is different about the whole page layout of the Register page and the other pages?

Eyrim commented on src/main/resources/static/css/application.css:
```diff
@@ -45,8 +45,50 @@
+display:flex;
+flex-direction:column;
+align-items: center;
+
```
Swap line 85 and 86

```suggestion
}

```

AndrewYHuang commented on gradlew.bat:
Why has this file changed?

AndrewYHuang commented on gradlew:
Why has this file changed?

AndrewYHuang commented on gradle/wrapper/gradle-wrapper.jar:
Why has this file changed?

AndrewYHuang commented on gradle/wrapper/gradle-wrapper.properties:
Why has this file changed?

Eyrim replied:
We're having gradle issues at the moment where gradle is changing itself, haven't found a workaround yet we just need to unstage it but we keep forgetting

AndrewYHuang replied:
Cool beans - I'll have a look at what's happening and make a fix to the main branch

Eyrim commented on src/main/resources/static/css/application.css:
```diff
@@ -45,8 +45,50 @@
+}
+
+
+.input-container
```
Braces on the same line as the rule with a space after the rule name and the brace

```suggestion
.input-container {
```

Eyrim reviewed (commented):
HTML looks fine, CSS needs formatting changes imo (I didn't add a comment for every instance of each issue be aware)

CadfanWilliams committed:
Code Cleanup

- Reformatted consistently
- Cleaned up blank spaces
- removed unused attributes
- reused "page-container"

CadfanWilliams committed:
Code Cleanup 2

- Forgot to remove this

AndrewYHuang commented on src/main/resources/templates/register.html:
```diff
@@ -1,33 +1,33 @@
+            </div>
+        </div>
+    </form>
+</section>
```
Please also indent these lines within section

AndrewYHuang commented on src/main/resources/templates/register.html:
```diff
@@ -1,33 +1,33 @@
+        </div>
+    </form>
+</section>
+</body>
```
Please indent these lines within body

AndrewYHuang commented on src/main/resources/static/css/application.css:
```diff
@@ -45,8 +41,37 @@
+
+.register-textfields {
+    height:2rem;
+}
```
Please include one newline at the bottom (note the angry red ⛔ symbol) :') This is because text files on *nix platforms are strictly only text files if and only if they end with a `\n` at the end

AndrewYHuang reviewed (changes requested):
Looks a lot better, just a few more nitpics from me

CadfanWilliams committed:
Code Cleanup 3

Removed unnecessary height attribute

CadfanWilliams committed:
Code Reformatting

 - Corrected indenting

AndrewYHuang reviewed (approved):
LGTM!

CadfanWilliams committed:
Merge branch 'main' into ws-004
